### PR TITLE
KNOX-2833 - Ozone integration for Apache Knox

### PR DESCRIPTION
**Pull request by debiswal**
Apache JIRA ID: [https://issues.apache.org/jira/browse/KNOX-2833] 

##  changes  proposed in this PR:
1. integration of Ozone Manager (OM) in KNOX UI
2. integration of Recon in KNOX UI
3. integration of SCM in KNOX UI
OM,Recon & SCM are  three service endpoints for Ozone Service.

## How  this patch tested?
By Accessing knox gateway home page using below cluster url  then try to access the respective pages from cdp-proxy for Ozone Manager(OM), Recon & Stoarge Container Manager (SCM) 
KNOX cluster url:
[https://ozone-knox-1.ozone-knox.root.hwx.site:8443/gateway/homepage/home/](url)

### CODE REVIEW

**Review by smolnar82 — CHANGES_REQUESTED** [review]
@debiswal - please address my review comments below.
Before we merge your changes, I'd like to ask you to write separate unit test classes for the newly introduced service model generators (there are plenty of samples under the `src/test/java` folder in the `gateway-discovery-cm` module.

Thanks!

File: gateway-discovery-cm/src/main/java/org/apache/knox/gateway/topology/discovery/cm/model/ozone/OzoneServiceModelGenerator.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+ */
+package org.apache.knox.gateway.topology.discovery.cm.model.ozone;
+
+import com.cloudera.api.swagger.client.ApiException;
+import com.cloudera.api.swagger.model.ApiConfigList;
+import com.cloudera.api.swagger.model.ApiRole;
+import com.cloudera.api.swagger.model.ApiService;
+import com.cloudera.api.swagger.model.ApiServiceConfig;
+import org.apache.knox.gateway.topology.discovery.cm.ServiceModel;
+import org.apache.knox.gateway.topology.discovery.cm.ServiceModelGeneratorHandleResponse;
```
Comment (zeroflag) [inline gateway-discovery-cm/src/main/java/org/apache/knox/gateway/topology/discovery/cm/model/ozone/OzoneServiceModelGenerator.java:25]: Please remove this unused import.

**Review by zeroflag — CHANGES_REQUESTED** [review]

**Comment by smolnar82** [comment]
Thanks, @debiswal for addressing my direct review comments. There is one more I mentioned above:

> Before we merge your changes, I'd like to ask you to write separate unit test classes for the newly introduced service model generators (there are plenty of samples under the src/test/java folder in the gateway-discovery-cm module.

Without the new test classes, we cannot merge your changes.

**Comment by debiswal** [comment]
> Thanks, @debiswal for addressing my direct review comments. There is one more I mentioned above:
> 
> > Before we merge your changes, I'd like to ask you to write separate unit test classes for the newly introduced service model generators (there are plenty of samples under the src/test/java folder in the gateway-discovery-cm module.
> 
> Without the new test classes, we cannot merge your changes.

@smolnar82 -  Noticed your comments have started looking into it already :)

**Comment by debiswal** [comment]
Hi @smolnar82 Thank you for review  would like to inform   SSL_ENABLED is defined at the role level in CM, so it should be defined on the role level in the model generator as well & earlier it was set in ServiceProperty which is not correct & now it is set to  RoleProperty  .we have changed where we get the property but we have not changed it where we store the property. 
The current patch is tested as well in knox cluster.

**Comment by smolnar82** [comment]
@debiswal - please fix the build failure and submit a new PS. Thanks!

**Review by zeroflag — APPROVED** [review]

**Review by smolnar82 — APPROVED** [review]

**Pull request merged**